ql/instruments/makevanillaswap.cpp

```diff
@@ -156,6 +156,7 @@ namespace QuantLib {
 
         Rate usedFixedRate = fixedRate_;
         if (fixedRate_ == Null<Rate>()) {
+            QL_REQUIRE(nominal_ != 0.0, "Cannot determine fair rate for zero nominal.");
             VanillaSwap temp(type_, nominal_,
```

Comment:
The problem here can be avoided by using 100 instead of `nominal_` for the `temp` swap.

Reply:
Yep, even better. I will adjust the PR.